src/index.ts

```diff
@@ -431,26 +437,18 @@ export class Upload extends Pumpify {
     return res;
   }
 
-  private async getRequestStream(reqOpts: r.OptionsWithUrl):
-      Promise<r.Request> {
+  private async getRequestStream(reqOpts: GaxiosOptions):
+      Promise<GaxiosResponse> {
     try {
       if (this.userProject) {
-        reqOpts.qs = reqOpts.qs || {};
-        reqOpts.qs.userProject = this.userProject;
+        reqOpts.params = reqOpts.params || {};
+        reqOpts.params.userProject = this.userProject;
       }
-      const authHeaders =
-          await this.authClient.getRequestHeaders(reqOpts.url as string);
-      reqOpts.headers = Object.assign({}, reqOpts.headers, authHeaders);
-      const requestStream = request(reqOpts);
-      requestStream.on('error', this.destroy.bind(this));
-      requestStream.on('response', this.onResponse.bind(this));
-      requestStream.on('complete', (resp) => {
-        const body = resp.body;
-        if (body && body.error) this.destroy(body.error);
+
+      return this.authClient.request(reqOpts).then(res => {
```

Comment:
I think you might want to do 

```js
return await this.authClient.request();
```

Otherwise I believe any request errors will not be caught by the `try/catch`

Reply:
Would this work as well? https://github.com/googleapis/gcs-resumable-upload/pull/174#discussion_r252008963

Reply:
Yep! Missed that comment :)